google/resource_compute_instance_test.go

```diff
@@ -1063,6 +1064,51 @@ func TestAccComputeInstance_hostname(t *testing.T) {
 	})
 }
 
+func TestAccComputeInstance_shieldedVmConfig(t *testing.T) {
+	t.Parallel()
+
+	var instance computeBeta.Instance
+	instanceName := fmt.Sprintf("terraform-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeInstance_shieldedVmConfig(instanceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeInstanceExists("google_compute_instance.foobar", &instance),
+					testAccCheckComputeInstanceHasShieldedVmConfig(&instance, true, true, true),
```

Comment:
I'm curious what happens if we include the block with two different shapes; mixed false+true values, and an entirely false block. Based on your defaults, we can define:

```
	shielded_vm_config {
		enable_secure_boot          = true
		enable_vtpm                 = true
	}
```

```
	shielded_vm_config {}
```

Do you mind adding instances with definitions like these in the `testAccComputeInstance_shieldedVmConfig` config?